api/src/DataPersister/PeriodDataPersister.php

```diff
@@ -19,9 +22,56 @@ public function __construct(
         );
     }
 
+    public function beforeCreate($data): BaseEntity {
+        static::updateDaysAndScheduleEntries($data);
+
+        return $data;
+    }
+
+    public function beforeUpdate($data): BaseEntity {
+        $orig = $this->em->getUnitOfWork()->getOriginalEntityData($data);
+
+        static::updateDaysAndScheduleEntries($data, $orig);
+
+        return $data;
+    }
+
     public function beforeRemove($data): ?BaseEntity {
         $this->validator->validate($data, ['groups' => ['delete', 'Period:delete']]);
 
         return null;
     }
+
+    public static function updateDaysAndScheduleEntries(Period $period, array $orig = null) {
+        $length = $period->getPeriodLength();
+        $days = $period->getDays();
+
+        $minDelta = 0;
+        if (null != $orig) {
+            $minDelta = $orig['start']->getTimestamp() - $period->start->getTimestamp();
+            $minDelta = floor($minDelta / 60);
+        }
```

Comment:
Ich musste lange überlegen bis ich gemerkt habe dass das nicht das "minimale Delta" heissen soll.

```suggestion
        $deltaMinutes = 0;
        if (null != $orig) {
            $deltaMinutes = $orig['start']->getTimestamp() - $period->start->getTimestamp();
            $deltaMinutes = floor($deltaMinutes / 60);
        }
```

Reply:
Vielleicht diese Berechnung ins `if (!$period->moveScheduleEntries) { ... }` verschieben? Brauchts ja erst dort, und dann wäre es näher an der eigentlichen Benutzung dran.